Can't compile the fresh custom Vagrant build

Just tried to make a fresh custom build using Vagrant environment, unfortunately it ends with the following error:  

....
....
default: Compiling .pio/build/custom_IR_ESP8266_4M1M/FrameworkArduino/Tone.cpp.o
    default: Compiling .pio/build/custom_IR_ESP8266_4M1M/FrameworkArduino/TypeConversion.cpp.o
    default: Compiling .pio/build/custom_IR_ESP8266_4M1M/FrameworkArduino/Updater.cpp.o
    default: Compiling .pio/build/custom_IR_ESP8266_4M1M/FrameworkArduino/WMath.cpp.o
    default: /home/vagrant/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/WMath.cpp: In function 'unsigned int makeWord(unsigned char, unsigned char)':
    default: /home/vagrant/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/WMath.cpp:84:55: error: new declaration 'unsigned int makeWord(unsigned char, unsigned char)'
    default:  unsigned int makeWord(unsigned char h, unsigned char l) {
    default:                                                        ^
    default: In file included from include/../src/Custom.h:7:0,
    default:                  from include/ESPEasy_config.h:94,
    default:                  from <command-line>:0:
    default: /home/vagrant/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Arduino.h:259:10: error: ambiguates old declaration 'uint16_t makeWord(byte, byte)'
    default:  uint16_t makeWord(byte h, byte l);
    default:           ^
    default: *** [.pio/build/custom_IR_ESP8266_4M1M/FrameworkArduino/WMath.cpp.o] Error 1
    default: ========================= [FAILED] Took 652.57 seconds =========================
    default: Environment             Status    Duration
    default: ----------------------  --------  ------------
    default: custom_IR_ESP8266_4M1M  FAILED    00:10:52.570
    default: ==================== 1 failed, 0 succeeded in 00:10:52.570 ====================
    default: Error: Option '-e' requires an argument.
    default: Error: Option '-e' requires an argument.
    default: Error: Option '-e' requires an argument.
    default: Error: Option '-e' requires an argument.
    default: Error: Option '-e' requires an argument.
    default: Created /tmp/tmp.me02qYkyTl
    default: Created /tmp/tmp.me02qYkyTl/ESPEasy_collect_dist
    default: ### Creating zip archives
    default:
    default: zip error: Nothing to do! (/home/vagrant/GitHub/letscontrolit/ESPEasy/ESPEasy_ELF_files_20220821_vagrant.zip)
    default: ### Created ESPEasy_ELF_files_20220821_vagrant.zip
    default: cp:
    default: cannot stat 'build_output/bin/*'
    default: : No such file or directory
    default: ### Created ESPEasy_ESP82xx_20220821_vagrant.zip
    default: cp:
    default: cannot stat 'build_output/bin/*ESP32*'
    default: : No such file or directory

--- a/src/Custom-sample.h
+++ b/src/Custom-sample.h
@@ -1,8 +1,6 @@
 #ifndef ESPEASY_CUSTOM_H
 #define ESPEASY_CUSTOM_H
 
-#include <Arduino.h>
-
 /*
     To modify the stock configuration without changing the EspEasy.ino file :
     1) rename this file to "Custom.h" (It is ignored by Git)

--- a/src/src/Helpers/Scheduler.cpp
+++ b/src/src/Helpers/Scheduler.cpp
@@ -915,9 +915,7 @@ void ESPEasy_Scheduler::process_gpio_timer(unsigned long id) {
   uint8_t pinNumber     = static_cast<uint8_t>((id >> 8) & 0xFF);
   uint8_t pinStateValue = static_cast<uint8_t>((id >> 16) & 0xFF);
 
-  bool success = true;
-
-  uint8_t pluginID;
+  uint8_t pluginID = PLUGIN_GPIO;
 
   switch (GPIOType)
   {
@@ -925,39 +923,37 @@ void ESPEasy_Scheduler::process_gpio_timer(unsigned long id) {
       GPIO_Internal_Write(pinNumber, pinStateValue);
       pluginID = PLUGIN_GPIO;
       break;
-    case GPIO_TYPE_MCP:
 #ifdef USES_P009
+    case GPIO_TYPE_MCP:
       GPIO_MCP_Write(pinNumber, pinStateValue);
       pluginID = PLUGIN_MCP;
-#endif
       break;
-    case GPIO_TYPE_PCF:
+#endif
 #ifdef USES_P019
+    case GPIO_TYPE_PCF:
       GPIO_PCF_Write(pinNumber, pinStateValue);
       pluginID = PLUGIN_PCF;
-#endif
       break;
+#endif
     default:
-      success = false;
+      return;
   }
 
-  if (success) {
-    const uint32_t key = createKey(pluginID, pinNumber);
+  const uint32_t key = createKey(pluginID, pinNumber);
 
-    // WARNING: operator [] creates an entry in the map if key does not exist
-    portStatusStruct tempStatus = globalMapPortStatus[key];
+  // WARNING: operator [] creates an entry in the map if key does not exist
+  portStatusStruct tempStatus = globalMapPortStatus[key];
 
-    tempStatus.mode    = PIN_MODE_OUTPUT;
-    tempStatus.command = 1; // set to 1 in order to display the status in the PinStatus page
+  tempStatus.mode    = PIN_MODE_OUTPUT;
+  tempStatus.command = 1; // set to 1 in order to display the status in the PinStatus page
 
-    if (tempStatus.state != pinStateValue) {
-      tempStatus.state        = pinStateValue;
-      tempStatus.output       = pinStateValue;
-      tempStatus.forceEvent   = 1;
-      tempStatus.forceMonitor = 1;
-    }
-    savePortStatus(key, tempStatus);
+  if (tempStatus.state != pinStateValue) {
+    tempStatus.state        = pinStateValue;
+    tempStatus.output       = pinStateValue;
+    tempStatus.forceEvent   = 1;
+    tempStatus.forceMonitor = 1;
   }
+  savePortStatus(key, tempStatus);
 }
 
 /*********************************************************************************************\

